### PR TITLE
(BREAKING) service queue customPatientChartUrl config to have working default value

### PR DESCRIPTION
**Pull request by chibongho**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
In the Service Queues app, the `customPatientIdUrl` config defaulted to empty string, which results in 2 issues: 
- broken "Profile" links in `<ActiveVisitsTable>` (see screenshot below)
- the `usePatientId ` hook makes requests to `"${customPatientIdUrl}=${patientUuid}"`, resulting in incorrect URL `"/=${patientUuid}"`

This PR adds a working default config value so that the links work without additional config (like in dev3.openmrs.org). This value follows the same format as the corresponding config in `esm-appointments-app`; see [this PR](https://github.com/openmrs/openmrs-esm-patient-management/pull/891).

After consulting with @ojwanganto, I also removed the "Profile" link and just have the link with the patient name (first column in table) use the customPatientIdUrl

@ojwanganto @donaldkibet @makombe Please note that this is likely a BREAKING CHANGE. The Palladium `customPatientIdUrl` config was likely set to `"${openmrsBase}/coreapps/clinicianfacing/patient.page?patientId"` to point to the O2 patient chart. If so, it will now need to be `"${openmrsBase}/coreapps/clinicianfacing/patient.page?patientId=${patientUuid}"`. To use the O3 patient chart, just leave this value undefined in the config. 

## Screenshots

image

### CODE REVIEW

File: packages/esm-service-queues-app/src/active-visits/active-visits-table.component.tsx [hunk 1]
```diff
@@ -361,9 +361,6 @@ function ActiveVisitsTable() {
                           <TableCell className="cds--table-column-menu">
                             <EditMenu queueEntry={visitQueueEntries?.[index]} closeModal={() => true} />
                           </TableCell>
-                          <TableCell className="cds--table-column-menu">
-                            <OpenChartMenu patientUuid={visitQueueEntries?.[index]?.patientUuid} />
-                          </TableCell>
```
Comment (mogoodrich) [inline packages/esm-service-queues-app/src/active-visits/active-visits-table.component.tsx:366]: I assume this is the "Profile" link you were mentioned removing?

**Review by mogoodrich — APPROVED** [review]
One small question, but generally LGTM, thanks @chibongho !

File: packages/esm-service-queues-app/src/active-visits/active-visits-table.component.tsx (hunk 1 above)
Comment (chibongho) [reply packages/esm-service-queues-app/src/active-visits/active-visits-table.component.tsx:366]: yep!

**Review by mseaton — APPROVED** [review]
LGTM

File: packages/esm-service-queues-app/src/config-schema.ts [hunk 2]
```diff
@@ -122,13 +122,9 @@ export const configSchema = {
...
-    _default: '',
-    _description: 'Custom URL to load patient chart',
-  },
-  customPatientIdUrl: {
-    _type: Type.String,
-    _default: '',
-    _description: 'Custom URL to fetch patient with id',
+    _default: '${openmrsSpaBase}/patient/${patientUuid}/chart',
+    _description:
+      'Template URL that will be used when clicking on the patient name in the queues table. Available argument: patientUuid',
```
Comment (brandones) [inline packages/esm-service-queues-app/src/config-schema.ts:127]:
```suggestion
      'Template URL that will be used when clicking on the patient name in the queues table. Available argument: patientUuid',
    _validators: [validators.isUrlWithTemplateParameters(['patientUuid']),
```

You'll need to add the import line. Shame that there's no validators used in this config schema yet but it's never too late to start :)

Comment (ibacher) [reply packages/esm-service-queues-app/src/config-schema.ts:127]: Possibly worth noting that both `${openmrsBase}` and `${openmrsSpaBase}` are also available (default provided by the underlying API; but having this front and center is helpful for people trying to customize the URL, e.g., you can do `${openmrsBase}/coreapps/clinicianfacing/patient.page?patientId=${patientUuid}`.

Comment (brandones) [reply packages/esm-service-queues-app/src/config-schema.ts:127]: This information is front and center by default in the implementer tools... but I understand the implementer tools are at present somewhat orphaned. Though I think they are very relevant to PIH's needs in particular.

Comment (chibongho) [reply packages/esm-service-queues-app/src/config-schema.ts:127]: I'll add those. I didn't know about the `_validators` field. Should we try to type the `configschema` for each app more? I see this here:

https://github.com/openmrs/openmrs-esm-core/blob/b52bba717365134541b4fb7a719f5e5040c4afb7/packages/esm-config/src/types.ts

Comment (brandones) [reply packages/esm-service-queues-app/src/config-schema.ts]: Yeah, use validators when it makes sense to. Please read the [config system documentation](https://o3-docs.openmrs.org/docs/configuration-system).

**Review by donaldkibet — APPROVED** [review]

**Comment by denniskigen** [comment]
E2E tests are passing now. This is probably good to go, @chibongho.

File: packages/esm-service-queues-app/src/active-visits/active-visits-table.component.tsx [hunk 3]
```diff
@@ -168,7 +166,9 @@ function ActiveVisitsTable() {
       ...entry,
       name: {
         content: (
-          <ConfigurableLink to={`\${openmrsSpaBase}/patient/${entry.patientUuid}/chart`}>{entry.name}</ConfigurableLink>
+          <ConfigurableLink to={customPatientChartUrl} templateParams={{ patientUuid: entry.patientUuid }}>
+            {entry.name}
+          </ConfigurableLink>
```
Comment (denniskigen) [inline packages/esm-service-queues-app/src/active-visits/active-visits-table.component.tsx:171]: 🚀

**Review by denniskigen — APPROVED** [review]

File: packages/esm-appointments-app/src/config-schema.ts [hunk 4]
```diff
@@ -68,9 +68,11 @@ export const configSchema = {
   },
   customPatientChartUrl: {
     _type: Type.String,
-    _description:
-      'Template URL that will be used when clicking on the patient name in the appointment list. Available argument: patientUuid',
+    _description: `Template URL that will be used when clicking on the patient name in the queues table. 
+      Available argument: patientUuid, openmrsSpaBase, openBase
```
Comment (denniskigen) [inline packages/esm-appointments-app/src/config-schema.ts:72]:
```suggestion
      Available arguments: patientUuid, openmrsSpaBase, openBase
```

File: packages/esm-service-queues-app/src/config-schema.ts [hunk 5]
```diff
@@ -122,13 +122,11 @@ export const configSchema = {
...
-    _default: '',
-    _description: 'Custom URL to load patient chart',
-  },
-  customPatientIdUrl: {
-    _type: Type.String,
-    _default: '',
-    _description: 'Custom URL to fetch patient with id',
+    _default: '${openmrsSpaBase}/patient/${patientUuid}/chart',
+    _description: `Template URL that will be used when clicking on the patient name in the queues table. 
+      Available argument: patientUuid, openmrsSpaBase, openBase
```
Comment (denniskigen) [inline packages/esm-service-queues-app/src/config-schema.ts:127]:
```suggestion
      Available arguments: patientUuid, openmrsSpaBase, openBase
```

**Pull request merged**